### PR TITLE
Fix #7358: Non-fusion BattleMeks now generate movement heat

### LINKED ISSUE #7358: [Issue] Non-Fusion BattleMechs movement heat is incorrect
### Brief Description *

TacOps Advanced Units and Equipment, page 120 notes, among other things: 

"Non-fusion-powered BattleMechs generate heat in the same manner as BattleMechs (including heat for Walking and Running movement."

Currently, Fuel Cell and ICE powered *Battle*Mechs generate no heat when walking or running. Fission based mechs are generating heat correctly.

I have attached MTF files of a fission and fuel cell BattleMech for testing convenience. An ICE BattleMech, the Raider, is available in the standard unit set.  

### 3. Steps to Reproduce

1. There is only one canon non-fusion BattleMech, the Raider (and Raider Mk. II)
2. Select a variant of this mech and start a match.
3. Observe that no heat is generated from movement.

### Attach Files

[Testers.zip](https://github.com/user-attachments/files/21695733/Testers.zip)

### Severity *

Medium

### MegaMek Suite Version *

MekHQ-0.50.07-nightly-2025-08-01

### Operating System *

Windows 11

### Java Version *

Adoptium 17.0.11

### Final Verification

- [x] I confirm this is a single, unique issue that hasn't been reported before
- [x] I have included all necessary information and files to help reproduce this issue
- [x] I have asked on MegaMek Discord about this issue
- [x] I have confirmed this issue is being opened on the correct repository: MegaMek, MegaMekLab, or MekHQ

### PR DESCRIPTION
**Pull request by HammerGS**
Per TacOps pg 85, ICE and Fuel Cell powered BattleMeks generate movement heat (1/2/3 for walk/run/sprint) despite not having fusion engines.

  Changes:
  - ICE/Fuel Cell Meks now generate proper movement heat
  - Refactored Engine movement heat methods for clarity
  - Fixed bug where non-Mek entities could generate movement heat
  - Added 25 unit tests for non-fusion engine mechanics

  Verified ICE engine crit behavior:
  - Engine crits do NOT generate additional heat (per TacOps)
  - Engine crits trigger explosion check at TN 10/7/4 for 1/2/3 hits

  🤖 Generated with [Claude Code](https://claude.com/claude-code)

  Co-Authored-By: Claude Opus 4.5 <noreply@anthropic.com>

  ---
  PR Description

  ## Summary
  - Fix ICE/Fuel Cell BattleMeks not generating movement heat per TacOps pg 85
  - Add comprehensive unit tests for non-fusion engine mechanics
  - Fix edge case where vehicles incorrectly generated movement heat

  Fixes #7358

  ## Changes
  - `Engine.java`: Refactored `getWalkHeat()`, `getRunHeat()`, `getSprintHeat()` to properly handle ICE/Fuel Cell Meks and exclude non-Mek entities
  - `NonFusionEngineTest.java`: 25 new behavioral tests covering movement heat, engine type detection, and SCM interactions

  ## Test plan
  - [x] 25 unit tests passing
  - [x] ICE Mek walking generates 1 heat
  - [x] ICE Mek running generates 2 heat
  - [x] ICE Mek sprinting generates 3 heat
  - [x] ICE Vehicle generates 0 movement heat
  - [x] Engine crit on ICE Mek triggers explosion check (TN 10), no heat generated

### CODE REVIEW

File: megamek/unittests/megamek/common/NonFusionEngineTest.java [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+            assertEquals(4, engine.getWalkHeat(mek),
+                  "XXL engine should generate 4 heat when walking (2x normal)");
+        }
+
+        @Test
+        @DisplayName("XXL engine generates 6 heat when running (2x normal)")
+        void xxlEngineRunHeatIsDoubled() {
+            Mek mek = mock(Mek.class);
+            when(mek.hasWorkingSCM()).thenReturn(false);
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/unittests/megamek/common/NonFusionEngineTest.java]: Fixed

File: megamek/unittests/megamek/common/NonFusionEngineTest.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+
+    @Nested
+    @DisplayName("XXL Engine Heat Tests")
+    class XXLEngineHeatTests {
+
+        @Test
+        @DisplayName("XXL engine generates 4 heat when walking (2x normal)")
+        void xxlEngineWalkHeatIsDoubled() {
+            Mek mek = mock(Mek.class);
+            when(mek.hasWorkingSCM()).thenReturn(false);
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply megamek/unittests/megamek/common/NonFusionEngineTest.java]: Fixed
